features/dd-sdk-android-rum/src/test/kotlin/com/datadog/android/rum/internal/metric/FakeInternalLogger.kt

```diff
@@ -0,0 +1,57 @@
+/*
+ * Unless explicitly stated otherwise all files in this repository are licensed under the Apache License Version 2.0.
+ * This product includes software developed at Datadog (https://www.datadoghq.com/).
+ * Copyright 2016-Present Datadog, Inc.
+ */
+
+package com.datadog.android.rum.internal.metric
+
+import com.datadog.android.api.InternalLogger
+import com.datadog.android.core.metrics.PerformanceMetric
+import com.datadog.android.core.metrics.TelemetryMetricType
+
+class FakeInternalLogger : InternalLogger {
```

Comment:
I think with a "fake" internal logger, I can easier intercept the telemetry event in unit test

Reply:
you can also intercept them by using mock + argument captors, like (here is an example if only single argument needs to be captured; for multiple arguments syntax will be slightly different):

```
val mockLogger = mock<InternalLogger>()

argumentCaptor<FooBar>() {
  
  verify(mockLogger).sendMetric(capture(),...)
  assertThat(lastValue).isEqualTo(...)
 
}

```

We have multiple sites where we use mocks + captors in our tests. But it is not blocking, you can keep `FakeInternalLogger` if you want.

Reply:
Yes, actually I hesitated between mock and fake, I think fake here is simpler and it can meet what I need in the unit test, so let's keep it for the moment